src/main/java/com/basho/riak/client/core/query/timeseries/TableDefinition.java

```diff
@@ -0,0 +1,163 @@
+package com.basho.riak.client.core.query.timeseries;
+
+import java.util.*;
+
+/**
+ * Holds a definition for a Table in Time Series Riak.
+ * Immutable once created.
+ * @author Alex Moore <amoore at basho dot com>
+ * @since 2.0.4
+ */
+public class TableDefinition
+{
+    private final String tableName;
+    private final LinkedHashMap<String, FullColumnDescription> fullColumnDescriptions = new LinkedHashMap<>();
+    private final ArrayList<FullColumnDescription> partitionKeys = new ArrayList<>();
```

Comment:
Since the whole instance is immutable from the user perspective, there shouldn't be any race conditions.  On the other hand,  as we are not expecting that user will execute describe table each second, I think that we can leave this as is.